cli/command/container/create.go

```diff
@@ -286,11 +343,30 @@ func createContainer(ctx context.Context, dockerCli command.Cli, containerCfg *c
 	if warn := localhostDNSWarning(*hostConfig); warn != "" {
 		response.Warnings = append(response.Warnings, warn)
 	}
+
+	containerID = response.ID
 	for _, w := range response.Warnings {
 		_, _ = fmt.Fprintln(dockerCli.Err(), "WARNING:", w)
 	}
-	err = containerIDFile.Write(response.ID)
-	return response.ID, err
+	err = containerIDFile.Write(containerID)
```

Comment:
Slightly related; looks like we may be discarding this error if failing to write to a container-id file failed, and something fails below.

Reply:
We return it below but first try to resolve and write the creds.

Reply:
Yeah, it would likely be a corner case; I was looking at these lines, where we don't return the containerID, and don't include the error about the container id-file being written.


```bash
if err := copyDockerConfigIntoContainer(ctx, dockerCli.Client(), containerID, dockerConfigPathInContainer, newConfig); err != nil {
	return "", fmt.Errorf("injecting docker config.json into container failed: %w", err)
}
```

We could do an `errors.Join` or something along those lines, but not sure if that's super great, so let's keep that for follow-ups.

Doing the "copy to container" as part of `docker start` was something that crossed my mind, but not sure if we want to / can update the files before start (that may open up possible race-conditions as well).

Reply:
Do you know why we write the containerID to a file on create? That's a bit unexpected. I tried to keep a light foot through here.

Reply:
Oh! It's not written to a file by default (that function's signature is a bit misleading); it's only written to a file if the uses passes the `--cidfile=<some path>` option.

If the flag isn't set, nothing is written and the function always returns nil

I'm not a huge fan of that feature, but I think it was added for some users that wanted to automate things and/or start containers through some init system (not a good fit)